src/POData/ObjectModel/ModelDeserialiser.php

```diff
@@ -61,9 +62,9 @@ public function bulkDeserialise(ResourceEntityType $entityType, ODataEntry $payl
         // assemble data array
         $data = [];
         foreach ($payload->propertyContent->properties as $propName => $propSpec) {
-            if (array_key_exists($propName, $nonRelProp)) {
+            if (in_array($propName, $nonRelProp) || in_array(strtolower($propName), $nonRelProp)) {
```

Comment:
Now check for both raw and lowercased version, at least one of which we hopefully stored earlier